metadata_service/proxy/neo4j_proxy.py

```diff
@@ -592,14 +592,19 @@ def add_badge(self, *,
                   id: str,
                   badge_name: str,
                   category: str = '',
-                  resource_type: ResourceType = ResourceType.Table) -> None:
+                  resource_type: ResourceType,
+                  column_name: str = '') -> None:
```

Comment:
I think it would be better to have the callers be responsible for generating the correctly-formed id, rather than passing in `column_name`? That would generalize to badging other types of content better.

If adopted, would be sensible to have a helper method that generates the id from a column name and id.

An alternative would be to pass in a new type of object that encapsulates `ResourceType` and `id`, and is responsible for containing the state and logic needed to generate id.

wdyt?

Reply:
Through the API, the user would just do: `curl -X PUT https://{your metadata url}/table/{table key}/column/{column name}/badge/{badge name}?category={badge category}` they'd still need to include the column name somehow and we'd need to check that they want to badge a column vs a table. A `correctly-formed id` would still have to include the column name, and I think this is the most straightforward way to provide that. I can for sure add a helper function if we want to make it more clear that the `id` is changing though.

Reply:
For implementation clarification, do the API call formats need to be different for table and column badge? How does the API know when to use the different API classes? It looks like `ColumnDescriptionAPI` is also analogous to `TableDescriptionAPI` and takes in a `column_name`.

Reply:
`we're getting that info from the resource_type, right?` This is because the API call format is unique from the table badge and so we're able to figure out which class to call because of it.